src/main/java/nl/tudelft/ti2206/group9/util/CollisionMap.java

```diff
@@ -0,0 +1,233 @@
+            final CollisionHandler<C1, C2> handler) {
+        onCollision(collider, collidee, true, handler);
+    }
+
+    /**
+     * Adds a collision interaction to this collection.
+     *
+     * @param <C1> The collider type.
+     * @param <C2> The collidee (AbstractEntity that was moved into) type.
+     * @param collider The collider type.
+     * @param collidee The collidee type.
+     * @param symetric <code>true</code> if this collision is used for both
+     *                  C1 against C2 and vice versa;
+     *                  <code>false</code> if only for C1 against C2.
+     * @param handler The handler that handles the collision.
+     */
+    public <C1 extends AbstractEntity, C2 extends AbstractEntity> void
+    onCollision(
+            final Class<C1> collider, final Class<C2> collidee,
+            final boolean symetric, final CollisionHandler<C1, C2> handler) {
+        addHandler(collider, collidee, handler);
+        if (symetric) {
+            addHandler(collidee, collider, new InverseCollisionHandler<C2, C1>(
+                    handler));
+        }
+    }
+
+    /**
+     * Adds the collision interaction..
+     *
+     * @param collider The collider type.
+     * @param collidee The collidee type.
+     * @param handler The handler that handles the collision.
+     */
+    private void addHandler(final Class<? extends AbstractEntity> collider,
+                            final Class<? extends AbstractEntity> collidee,
+                            final CollisionHandler<?, ?> handler) {
+        if (!handlers.containsKey(collider)) {
+            handlers.put(
+                    collider,
+                    new HashMap<>());
+        }
+
+        final Map<Class<? extends AbstractEntity>, CollisionHandler<?, ?>
+                > map = handlers.get(collider);
+        map.put(collidee, handler);
+    }
+
+    /**
+     * Handles the collision between two colliding parties, if a suitable
+     * collision handler is listed.
+     *
+     * @param <C1> The collider type.
+     * @param <C2> The collidee (AbstractEntity that was moved into) type.
+     * @param collider The collider.
+     * @param collidee The collidee.
+     */
+    public <C1 extends AbstractEntity, C2 extends AbstractEntity> void
+    collide(final C1 collider, final C2 collidee) {
+        final Class<? extends AbstractEntity> colliderKey
+                = getMostSpecificClass(handlers, collider.getClass());
+        if (colliderKey == null) {
+            return;
+        }
+
+        final Map<Class<? extends AbstractEntity>, CollisionHandler<?, ?>> map
+                = handlers.get(colliderKey);
+        final Class<? extends AbstractEntity> collideeKey
+                = getMostSpecificClass(map, collidee.getClass());
+        if (collideeKey == null) {
+            return;
+        }
+
+        final CollisionHandler<C1, C2> collisionHandler
+                = (CollisionHandler<C1, C2>) map.get(collideeKey);
```

Comment:
Resolved, but the diff is not 'outdated' because the @suppresswarnings is put at the top of the method.